mlrun/datastore/azure_blob.py

```diff
@@ -71,7 +72,8 @@ def upload(self, key, src_path):
             with open(src_path, "rb") as data:
                 blob_client.upload_blob(data, overwrite=True)
         else:
-            remote_path = f"{self.endpoint}{key}"
+            key = key.strip("/")
+            remote_path = Path(self.endpoint, key).as_posix()
```

Comment:
I suggest moving these 2 lines to some helper method (`_convert_key_to_remote-path`) instead of repeating the logic over and over again

Reply:
Good point @Hedingber .  I've converted this to a private function.